src/JSInterop/Microsoft.JSInterop/src/DotNetDispatcher.cs

```diff
@@ -173,10 +173,12 @@ private static object[] ParseArguments(string methodIdentifier, string argsJson,
 
             // There's no direct way to say we want to deserialize as an array with heterogenous
             // entry types (e.g., [string, int, bool]), so we need to deserialize in two phases.
-            var jsonDocument = JsonDocument.Parse(argsJson);
+            JsonDocument jsonDocument = default;
             var shouldDisposeJsonDocument = true;
             try
             {
+                jsonDocument = JsonDocument.Parse(argsJson);
```

Comment:
I think you should be able to use the reader + serializer now instead of `JsonDocument` for better perf.

Reply:
Curious, how does it work?

Reply:
We could paired program a little if you'd like, or you could try it out on your own and use https://github.com/aspnet/AspNetCore/pull/10733 as a reference

Reply:
Tracking via https://github.com/aspnet/AspNetCore/issues/10988